zkstack_cli/crates/zkstack/src/commands/dev/commands/track_priority_txs.rs

```diff
@@ -0,0 +1,284 @@
+        .get_block_number()
+        .await
+        .expect("Failed to fetch latest block")
+        .as_u64();
+
+    let filter = Filter::new()
+        .address(diamond_proxy_address)
+        .topic0(topic)
+        .from_block(from_block)
+        .to_block(latest_block);
+
+    let priority_op_logs = l1_provider.get_logs(&filter).await?;
+    let mut result = vec![];
+
+    for log in priority_op_logs.into_iter().rev() {
+        let log = ethers_log_to_zk_log(log);
+
+        // All these expects are safe
+        let eth_tx_hash = log
+            .transaction_hash
+            .context("Historical log lacking transaction hash")?;
+        let block_number = log
+            .block_number
+            .context("Historical log lacking block number")?;
+        let l1_tx = L1Tx::try_from(log).context("Failed to decode priority log")?;
+
+        if let Some(expected_sender) = priority_op_sender {
+            if l1_tx.common_data.sender != expected_sender {
+                continue;
+            }
+        }
+
+        // To prevent rate limits on common providers, we need to `sleep` for a bit between frequent sequential calls
+        sleep(Duration::from_millis(200));
+        // Apparently, the block timestamp field is often not present, so we'll have to query it manually here
+        let block_timestamp = l1_provider
+            .get_block(block_number)
+            .await?
+            .context("Can not find block for Historical log")?
+            .timestamp
+            .as_u64();
+
+        let priority_tx_hash = l1_tx.hash();
+
+        let tx_status = l2_zk_client
+            .get_transaction_receipt(priority_tx_hash)
+            .await?;
+
+        match tx_status {
+            Some(receipt) => result.push(TxStatus {
+                block_timestamp,
+                eth_tx_hash,
+                priority_tx_hash,
+                state: if receipt.status.as_u64() == 0 {
+                    PriorityTxState::Fail
+                } else {
+                    PriorityTxState::Success
+                },
+            }),
+            None => result.push(TxStatus {
+                block_timestamp,
+                eth_tx_hash,
+                priority_tx_hash,
+                state: PriorityTxState::WaitingToProcess(latest_block - block_number.as_u64() + 1),
+            }),
+        }
+
+        if result.len() >= limit {
+            return Ok(result);
+        }
+    }
+
+    Ok(result)
+}
+
+fn display_statuses(statuses: Vec<TxStatus>) {
+    for status in statuses.iter() {
+        let (emoji, description) = match &status.state {
+            PriorityTxState::Success => ("✅", "Success".to_string()),
+            PriorityTxState::Fail => ("❌", "Failed".to_string()),
+            PriorityTxState::WaitingToProcess(conf) => {
+                ("⏳", format!("Waiting ({} confirmations)", conf))
+            }
+        };
+
+        println!("{} {} {:#?}", emoji, description, status.priority_tx_hash);
```

Comment:
ples use logger instead of println

Reply:
discussed in dm, this is intended in this case